compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -761,14 +761,10 @@ class Typer extends Namer
   def typedFunctionType(tree: untpd.Function, pt: Type)(implicit ctx: Context) = {
     val untpd.Function(args, body) = tree
     val (isImplicit, isErased) = tree match {
-      case tree: untpd.NonEmptyFunction =>
-        if (args.nonEmpty) (tree.mods.is(Implicit), tree.mods.is(Erased))
-        else {
-          ctx.error(FunctionTypeNeedsNonEmptyParameterList(tree.mods.is(Implicit), tree.mods.is(Erased)), tree.pos)
-          (false, false)
-        }
+      case tree: untpd.FunctionWithMods => (tree.mods.is(Implicit), tree.mods.is(Erased))
       case _ => (false, false)
     }
+    if (isErased && args.isEmpty) ctx.error(em"empty function cannot not be erased", tree.pos)
```

Comment:
This looks inconsistent. We do provide names for `ErasedFunction0` and `ErasedImplicitFunction0` and add them in scope, but we do not allow them to be used in function types.

* We could remove the start `ErasedFuctionN` and `ErasedImplicitFuctionN` at `N = 1`
* Or allow them, as a function type that is equivalent to `Funcion0` or `ImplicitFunction0`. We could even consider making them type aliases and warn the the `erased` keyword is useless.

Reply:
@nicolasstucki Agree we should start ErasedFunctionN at 1. Can you do the change and push it on top of this PR? Thanks!